scipy/interpolate/interpolate.py

```diff
@@ -2,17 +2,14 @@
 # Use the `scipy.interpolate` namespace for importing the functions
 # included below.
 
-import warnings
-from . import _interpolate
+from scipy._lib.deprecation import _sub_module_deprecation
 
 
 __all__ = [  # noqa: F822
     'BPoly',
     'BSpline',
     'NdPPoly',
     'PPoly',
-    'RectBivariateSpline',
-    'RegularGridInterpolator',
```

Comment:
Presumably these are things that don't exist anymore?

Reply:
Turns out this is a bit of an edge case, the contents are split across multiple private files

Reply:
TBH I don't fully understand the situation here, but I think these issues demonstrate that it would be great to have some testing that the content of our `__all__`'s is correct.